test/suite/test_hs33.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from helper import copy_wiredtiger_home
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest, time, threading
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+from wtthread import checkpoint_thread
+
+# test_hs33.py
+# Test that we can run recovery after crashing before the metadata is synced during recovery.
+# This simulates the scenario seen in WT-14376, where the eviction subsystem opened the history
+# store file before metadata recovery had completed.
+class test_hs33(wttest.WiredTigerTestCase, suite_subprocess):
+    conn_config = 'statistics=(all),log=(enabled=true)'
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def large_updates(self, session, uri, value, ds, nrows, timestamp=False):
+        cursor = session.open_cursor(uri)
+        for i in range(1, nrows):
+            if timestamp == True:
+                session.begin_transaction()
+            cursor.set_key(ds.key(i))
+            cursor.set_value(value)
+            self.assertEqual(cursor.update(), 0)
+            if timestamp == True:
+                session.commit_transaction('commit_timestamp=' + self.timestamp_str(i + 1))
+        cursor.close()
+
+    def large_modifies(self, session, uri, offset, ds, nrows, timestamp=False):
```

Comment:
This function is not used in this test; it can be removed.

Reply:
Removed https://github.com/wiredtiger/wiredtiger/pull/11857/commits/80ec911c73bea24a597b6b9fecdfa67ac071c65a